terraform/shared/modules/egress-proxy/test/validate.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/sh
+set -e
+
+# A quick script to validate that the deployed config is what we expect
+
+# Target the place where we think stuff is
+cf t -o $ORGNAME -s $SPACENAME > /dev/null 2>&1
+
+# Check that the egress app exists
+cf app ${APPNAME} > /dev/null 2>&1
+
+# Check that the egress app has the appropriate env vars set
+# TODO: Improve this to ensure the var content is as expected.
+cf env ${APPNAME} | grep -q PROXY_ALLOW
+cf env ${APPNAME} | grep -q PROXY_DENY
+cf env ${APPNAME} | grep -q PROXY_USERNAME
+cf env ${APPNAME} | grep -q PROXY_PASSWORD
+
+# Check that a network policy with appropriate values exists 
+# TODO: Improve this to use just one robust regexp (or use "cf curl" and "jq").
+cf network-policies | grep test | grep ${APPNAME} | grep 61443 | grep "$ORGNAME" | grep -q "$SPACENAME" 
+
+# Check that a user-provided service with the egress credentials exists
+cf service ${APPNAME}-creds > /dev/null 2>&1
```

Comment:
Add a `cf ssh` with a curl to test that the allow/deny acls are working

Reply:
I added a `cf ssh` test that uses `curl` and I can see that it's giving the results I expect. However, it's a little squirrely to test the output is what's expected because the return value is hard to extract over cf ssh. I'm too tired to work through that now so I'll need someone to help with it in the morning!